hphp/runtime/vm/jit/unique-stubs.cpp

```diff
@@ -1034,12 +1034,15 @@ TCA emitDecRefGeneric(CodeBlock& cb, DataBlock& data) {
   CGMeta meta;
 
   auto const start = vwrap(cb, data, meta, [] (Vout& v) {
+    v << vregrestrict{};
     v << stublogue{};
 
     auto const rdata = rarg(0);
     auto const rtype = rarg(1);
 
     auto const destroy = [&] (Vout& v) {
+      v << vregallow{};
```

Comment:
This should go after we construct the `PhysRegSaver` below.

Actually, I think it would be better if we just did the `v << vregallow{}` in PhysRegSaver's constructor after the pushes, and then the `v << vregrestrict{}` in the destructor, before the pops.  Then we can make it explicit in PhysRegSaver's documentation that this is what the PhysRegSaver "buys" us.